### ISSUE TITLE
Align tensorboard log curves when batch_size and number of gpus not equal.

### ISSUE THREAD

**Issue by LuoXin-s** [PyTorchLightning/pytorch-lightning#9470]
## 🚀 Feature

Align tensorboard log curves when batch_size and number of gpus not equal.

### Motivation

I may use different number of gpus to train my model, also different batch_size, but when I use self.log in LightningModule to log my loss/metrics curves not aligned with same horizontal axis.

### Additional context

______________________________________________________________________

#### If you enjoy Lightning, check out our other projects! ⚡

<sub>

- [**Metrics**](https://github.com/PyTorchLightning/metrics): Machine learning metrics for distributed, scalable PyTorch applications.

- [**Flash**](https://github.com/PyTorchLightning/lightning-flash): The fastest way to get a Lightning baseline! A collection of tasks for fast prototyping, baselining, finetuning and solving problems with deep learning

- [**Bolts**](https://github.com/PyTorchLightning/lightning-bolts): Pretrained SOTA Deep Learning models, callbacks and more for research and production with PyTorch Lightning and PyTorch

- [**Lightning Transformers**](https://github.com/PyTorchLightning/lightning-transformers): Flexible interface for high performance research using SOTA Transformers leveraging Pytorch Lightning, Transformers, and Hydra.

</sub>

**Comment by tchaton** [comment]
Dear @LuoXin-s,

Lightning aligns the curve on the number of `optimizer.step`. So this is not something Lightning will do.

Best,
T.C
